### DISCUSSION TITLE
Slide layout model - needs defining

### DISCUSSION THREAD

**Discussion by josephwright** [josephwright/ltx-talk#11] [General]
In `beamer`, the layout model is rather at odds with the typical LaTeX approach, e.g. the header area is non-existent and the top lines therefore sit in the 'body'. It would be good to work out how this should be approached. At present, my feeling is that we should keep more to the same approach as other documents: header and footer areas with a defined height using `geometry`, etc.

Related are the dimensions of slides: I'm minded to say that the height is always 100mm, with a border of (say) 2mm all round, so that the layout is rather more predictable that in `beamer`.

**Comment by josephwright** [comment]
@samcarter I'm particularly thinking you'll have insight here.

**Comment by samcarter** [comment]
I agree that a more traditional approach with head and footlines will probably be better. Maybe in the future combined with different pagestyles for title- , section pages etc.

Beamer also has this feature that it will shrink spaces once the content fills up the slide. I would ditch this in favour of a more predicable frame geometry.

It might also be worth thinking about if the height of the frame title can be subtracted from the text height. This would make it more convenient for scaling figures etc. 

---

> Related are the dimensions of slides: I'm minded to say that the height is always 100mm, with a border of (say) 2mm all round, so that the layout is rather more predictable that in beamer.

A fixed height makes sense imho, that's also what I choose to calculate the on-the-fly aspect ratios in beamer. Lines in a presentation are typically only partially filled, so changing the width (while keeping the height constant) often allows to change aspect rations without having to adapt the slide content.

**Comment by Skillmon** [comment]
This is going to be a boring discussion...

I totally agree on the fixed height with you two (which is also what I did in my own presentation class I derived from `beamer` for my presentations back in university). And 100pt makes calculations rather predictable and easier than 96pt, so I agree on that value as well (we just have to make sure to stay in a range in which typical font sizes make sense -- though 96pt has the advantage of being a multiple of 12).

On the geometry: I agree as well that we should provide a saner approach which results in valid `\textheight` and `\textwidth` for the content area. How exactly that should be achieved is a matter of implementation, `geometry` might be a good idea, but we could stick to a more box-oriented approach of coffins with the content-coffin being built last when the geometries are clear from the others. Another thing we could aim for is to have support a `\parshape` like mode for non-rectangular layouts, but I'm not sure it's worth it.

**Comment by josephwright** [comment]
> This is going to be a boring discussion...

Most likely: I just wanted to be sure we are all on the same page. More important is likely that we have a documented (and illustrated) explanation of the model used.

> I totally agree on the fixed height with you two (which is also what I did in my own presentation class I derived from `beamer` for my presentations back in university). And 100pt makes calculations rather predictable and easier than 96pt, so I agree on that value as well (we just have to make sure to stay in a range in which typical font sizes make sense -- though 96pt has the advantage of being a multiple of 12).

I assume you mean 100mm/96mm :) I can see the point of 96mm height as it's the classic one used in `beamer` as well as being a multiple of twelve. Of course, if we have a non-zero 'margin' then things are more tricky. I think I have 100mm mainly as I'm using 16:10 ratio slides, but that of course fails with 16:9 or other ratios - so perhaps sticking to 96mm is overall best.

> On the geometry: I agree as well that we should provide a saner approach which results in valid `\textheight` and `\textwidth` for the content area.

Again, I'm not surprised this is non-controversial, just wanted to be sure.

> How exactly that should be achieved is a matter of implementation, `geometry` might be a good idea, but we could stick to a more box-oriented approach of coffins with the content-coffin being built last when the geometries are clear from the others. Another thing we could aim for is to have support a `\parshape` like mode for non-rectangular layouts, but I'm not sure it's worth it.

I see the coffins idea, but unlike posters, I think a decent number of slides are pretty simply layout-wise, so I'm minded to go with `geometry`.

**Reply by josephwright** [reply]
I'm minded to go for 100mm as the fixed size. The reasoning is that in the end most slides will have some header/footer area that has to be accounted for, and we are not going to divide the body part up into 12. So something of roughly the right size that is memorable seems most sensible: I think 100mm works better on those grounds than 96mm, despite the history. Any objections before I make the change?

**Comment by Skillmon** [comment]
With a coffins based approach styling (and more complicated styles) are easier to implement. Sidebars are just a joined coffin for instance whereas in a `geometry` based approach placing them could turn out more complicated. Presentations tend to be a graphical thing much more than textual documents.

On the other hand, if one doesn't provide a `coffins` based approach the contents necessarily need to be more layered to achieve non-standard layouts (via `shipout/background` for instance), which might be an advantage in itself.

**Reply by josephwright** [reply]
Yes, I see that - partly comes down to how well we can handle tagging of material in boxes, I guess. (I wonder to what extend the content of sidebars even makes sense in that regard - things like a TOC in a sidebar on every slide, which is one design you see, feels to me like you'd not really want to know about it at all.)

**Reply by Skillmon** [reply]
Yes, when it comes to tagging the outline information in the sidebars seems like something you don't want to tag at all. But that can be achieved by providing a design level function to produce it that switches off tagging.

**Reply by josephwright** [reply]
Yup: but if we can establish which posts of the side need tags at all, that might help decide if coffins are a viable approach.

**Comment by cfr42** [comment]
I think my slides must look very different from everyone else's ...

It often matters a lot to me that I can use `plain` to switch off any header/footer. (I don't have a footer as such anyway.) 

Yes, this makes the layout inconsistent, but I think that is better than shrinking the size of the content which is often the only other option. A logical tableaux cannot be split across slides, for example. So either it gets smaller or it takes more (all) of the screen space. I try very hard not to reduce sizes for accessibility reasons, so I'd always use `plain` first.

Arguments in standard form often end up getting `plain`, too. There's a huge advantage to being able the fit the entire thing on a single slide. And they take up a great deal of space - both width and height. Even if I split across slides, I often use `plain` so I can fit at least the structure of the argument onto one slide. 

I apparently also use `handout` differently, because I often use multiple overlays in `handout` mode.  i've never liked the idea of dropping content from that version and overlays make it easy to avoid doing so.

**Reply by josephwright** [reply]
> It often matters a lot to me that I can use `plain` to switch off any header/footer. (I don't have a footer as such anyway.)

I don't see that as a problem: we clearly need the 'main area' to be tagged with a logical structure, so removal of the 'wallpaper' is no issue in the main.

The one thing to think about in this area is once again the frame title - that's something that I suspect many people (including me) don't routinely include but I suspect is needed for accessibility. (This leads to an aside: I've currently set up `\frametitle` as the way to give a title, but one could argue that all frames have to have a title, so it should be a mandatory argument - probably tricky with things like a title slide.)

> I apparently also use `handout` differently, because I often use multiple overlays in `handout` mode. i've never liked the idea of dropping content from that version and overlays make it easy to avoid doing so.

Multiple handouts are not necessarily an issue, though again one runs into more a a conceptual than code question. What is the 'right' amount of stuff on one slide before you need to have a second one. The classic example would be swapping out images: if you replace one with another, we can't 'flatten' to one representation easily, but the user probably doesn't want to wade through two entire slides just for that difference.

I guess for multiple handouts some form of examples is needed - I tend toward just just `<+->` lists or similar, so a single handout makes sense from multiple presentation slides.

**Reply by cfr42** [reply]
> I don't see that as a problem: we clearly need the 'main area' to be tagged with a logical structure, so removal of the 'wallpaper' is no issue in the main.

I typically use `plain` without a title. So some means of specifying a title for the structure without creating a visible title?  Though it isn't clear to me that a title is particularly useful when a slide contains, say, just a tableaux. 

> The one thing to think about in this area is once again the frame title - that's something that I suspect many people (including me) don't routinely include but I suspect is needed for accessibility. (This leads to an aside: I've currently set up `\frametitle` as the way to give a title, but one could argue that all frames have to have a title, so it should be a mandatory argument - probably tricky with things like a title slide.)

Mine usually do have a title unless I'm excluding it to save space or just to display a picture or something. Which is generally why I never know what to do with `\section`. 

> Multiple handouts are not necessarily an issue, though again one runs into more a a conceptual than code question. What is the 'right' amount of stuff on one slide before you need to have a second one. The classic example would be swapping out images: if you replace one with another, we can't 'flatten' to one representation easily, but the user probably doesn't want to wade through two entire slides just for that difference.

I never just swap images unless the images are the point. But I do sometimes swap one example for another, say. So the heading of the block or the definition stays the same, but a different example is shown. If you make that 2 slides, you get wobble, so I make it one and use `\begin{frame}<1-| handout:1-2>` or whatever. The other place I use it is if I want to use `\againframe`. So I might have part of a tree or part of a mindmap and then some other stuff and then a more complete version. Then  I have e.g. `\againframe<7-10| handout:3>{label}` or whatever.

> I guess for multiple handouts some form of examples is needed - I tend toward just just `<+->` lists or similar, so a single handout makes sense from multiple presentation slides.

Do you mean examples of how this gets used?

**Reply by josephwright** [reply]
> > I don't see that as a problem: we clearly need the 'main area' to be tagged with a logical structure, so removal of the 'wallpaper' is no issue in the main.
> 
> I typically use `plain` without a title. So some means of specifying a title for the structure without creating a visible title? Though it isn't clear to me that a title is particularly useful when a slide contains, say, just a tableaux.

I suspect we will need some input from accessibility experts here - quite what is 'needed' I'm not sure. I can see for example that if you have a run of slides about the same topic, a title on the first one then no titles later might be a reasonable structure. So probably the position is that, as we've set up, frame titles are optional.

> Mine usually do have a title unless I'm excluding it to save space or just to display a picture or something. Which is generally why I never know what to do with `\section`.

Well yes, this very much depends on how you approach your slides. I don't use `\section` for single talks, but I do if I have a single PDF for an entire lecture course - that way I get a TOC I can use to signpost.

**Reply by Skillmon** [reply]
> I can see for example that if you have a run of slides about the same topic, a title on the first one then no titles later might be a reasonable structure.

We could provide an environment for this use case, something like `\begin{framerange}{<title>}\begin{frame}[use-rangetitle]...\end{frame}\begin{frame}...\end{frame}...\end{framerange}` and get this tagged accordingly.

**Comment by wspr** [comment]
I agree “full slide” designs are not uncommon, which take over the entire space used for the title. 

It would seem sensible to me that if slide titles are needed for accessibility, that if one is not provided there is an implicit (invisible) “Slide XX” title used for tagging. 

The ramification is that the implicit title should be customisable, so

    \frametitle[tagonly]{…}

or similar might make sense.

**Comment by cfr42** [comment]
does it make a difference if slides are nested within a `Form` environment?

**Reply by Skillmon** [reply]
Are you referring to my `framerange` thingy? For tagging it might.

**Reply by cfr42** [reply]
> Are you referring to my `framerange` thingy? For tagging it might.

Nothing that specific. I just wondered whether or how surrounding the content by a `Form` environment would affect the structure. My slides pretty much always are inside a `Form` environment. 

Probably not important.

**Comment by josephwright** [comment]
I think we are resolved at the moment to go with a standard header/footer, with just the question of where we attach the margin boxes to deal with. That can be sorted in a dedicated discussion (my feeling is the header 'holds' these)).
